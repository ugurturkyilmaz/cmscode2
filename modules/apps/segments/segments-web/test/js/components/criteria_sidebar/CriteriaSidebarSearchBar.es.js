/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {cleanup, fireEvent, render} from '@testing-library/react';
import PropTypes from 'prop-types';
import React, {Component} from 'react';

import CriteriaSidebarSearchBar from '../../../../src/main/resources/META-INF/resources/js/components/criteria_sidebar/CriteriaSidebarSearchBar.es';

const SEARCH_BUTTON_TESTID = 'search-button';

const SEARCH_INPUT_TESTID = 'search-input';

class TestComponent extends Component {
	static propTypes = {
		initialValue: PropTypes.string,
	};

	static defaultProps = {
		initialValue: '',
	};

	constructor(props) {
		super(props);
		this.state = {
			value: props.initialValue,
		};
	}

	_handleChange = (value) => this.setState({value});

	render() {
		return (
			<CriteriaSidebarSearchBar
				onChange={this._handleChange}
				searchValue={this.state.value}
			/>
		);
	}
}

describe('CriteriaSidebarSearchBar', () => {
	afterEach(cleanup);

	it('renders', () => {
		const {asFragment} = render(
			<CriteriaSidebarSearchBar onChange={jest.fn()} />
		);

		expect(asFragment()).toMatchSnapshot();
	});

	it('renders with a blank search input with no search value', () => {
		const {getByTestId} = render(
			<CriteriaSidebarSearchBar onChange={jest.fn()} />
		);

		const searchInput = getByTestId(SEARCH_INPUT_TESTID);

		expect(searchInput.value).toEqual('');
	});

	it('renders with the value in the search input', () => {
		const {getByTestId} = render(
			<CriteriaSidebarSearchBar onChange={jest.fn()} searchValue="test" />
		);

		const searchInput = getByTestId(SEARCH_INPUT_TESTID);

		expect(searchInput.value).toEqual('test');
	});

	it('renders a button with a times icon when an input is entered', () => {
		const {getByTestId} = render(
			<CriteriaSidebarSearchBar onChange={jest.fn()} searchValue="test" />
		);

		const searchButton = getByTestId(SEARCH_BUTTON_TESTID);

		expect(searchButton).toMatchSnapshot();
	});

	it('clears the input when the times icon is clicked', () => {
		const {getByTestId} = render(<TestComponent initialValue="test" />);

		const searchButton = getByTestId(SEARCH_BUTTON_TESTID);
		const searchInput = getByTestId(SEARCH_INPUT_TESTID);

		expect(searchInput.value).toEqual('test');

		fireEvent.click(searchButton);

		expect(searchInput.value).toEqual('');
	});
});
