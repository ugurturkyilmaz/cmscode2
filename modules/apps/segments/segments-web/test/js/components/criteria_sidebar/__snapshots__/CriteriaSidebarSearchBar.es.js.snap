// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CriteriaSidebarSearchBar renders 1`] = `
<DocumentFragment>
  <div
    class="input-group"
  >
    <div
      class="input-group-item"
    >
      <input
        class="form-control input-group-inset input-group-inset-after"
        data-testid="search-input"
        placeholder="search"
        type="text"
        value=""
      />
      <div
        class="input-group-inset-item input-group-inset-item-after"
      >
        <button
          class="btn btn-unstyled"
          data-testid="search-button"
          type="button"
        >
          <svg
            class="lexicon-icon lexicon-icon-search"
            role="presentation"
          >
            <use
              xlink:href="#search"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CriteriaSidebarSearchBar renders a button with a times icon when an input is entered 1`] = `
<button
  class="btn btn-unstyled"
  data-testid="search-button"
  type="button"
>
  <svg
    class="lexicon-icon lexicon-icon-times"
    role="presentation"
  >
    <use
      xlink:href="#times"
    />
  </svg>
</button>
`;
